include/deal.II/fe/mapping_p_generic.h

```diff
@@ -252,7 +254,7 @@ class MappingPGeneric : public Mapping<dim, spacedim>
      * Constructor. The argument denotes the polynomial degree of the mapping
      * to which this object will correspond.
      */
-    InternalData(const unsigned int polynomial_degree);
+    InternalData(const FiniteElement<dim, spacedim> &fe);
```

Comment:
Is this a `FiniteElement` so that you know the element geometry?

Reply:
Yes: polynomials + geometric shape.